real_preprocessing/src/camera_pose.cpp

```diff
@@ -0,0 +1,263 @@
+    }
+    fout.close();
+  }
+
+  //solvePnP calculator given file number and tag position in file
+  Eigen::MatrixXd objTcam(int loc, int filenum)
+  {
+    YAML::Node tags_pix = YAML::LoadFile("detections_" + std::to_string(filenum) + ".yaml");
+    int bl_pix_x = tags_pix["detections"][loc]["corners"]["0"][0].as<int>();
+    int bl_pix_y = tags_pix["detections"][loc]["corners"]["0"][1].as<int>();
+    int br_pix_x = tags_pix["detections"][loc]["corners"]["1"][0].as<int>();
+    int br_pix_y = tags_pix["detections"][loc]["corners"]["1"][1].as<int>();
+    int tr_pix_x = tags_pix["detections"][loc]["corners"]["2"][0].as<int>();
+    int tr_pix_y = tags_pix["detections"][loc]["corners"]["2"][1].as<int>();
+    int tl_pix_x = tags_pix["detections"][loc]["corners"]["3"][0].as<int>(); 
+    int tl_pix_y = tags_pix["detections"][loc]["corners"]["3"][1].as<int>();
+    int tag_id = tags_pix["detections"][loc]["targetID"].as<int>();
+    double tag_size = tags_pix["detections"][loc]["size"][0].as<double>();
+
+    std::vector<cv::Point2d> img_pts;
+    std::vector<cv::Point3d> obj_pts;
+    cv::Mat obj_T_cam_rvec;
+    cv::Mat obj_T_cam_tvec;
+    cv::Mat rodrigues_rvec;
+    
+    img_pts.push_back(cv::Point2d(bl_pix_x, bl_pix_y));
+    img_pts.push_back(cv::Point2d(br_pix_x, br_pix_y));
+    img_pts.push_back(cv::Point2d(tr_pix_x, tr_pix_y));
+    img_pts.push_back(cv::Point2d(tl_pix_x, tl_pix_y));
+
+    //object corner specification
+    obj_pts.push_back(cv::Point3d(-(tag_size / 2), -(tag_size / 2), 0));
+    obj_pts.push_back(cv::Point3d((tag_size / 2), -(tag_size / 2), 0));
+    obj_pts.push_back(cv::Point3d((tag_size / 2), (tag_size / 2), 0));
+    obj_pts.push_back(cv::Point3d(-(tag_size / 2), (tag_size / 2), 0));
+
+    cv::Mat cam_matrix(3,3,CV_64F);
+    cv::Vec<float, 5> distCoeffs;
+    intrinsicLoad(cam_matrix, distCoeffs);
+
+    cv::solvePnP(obj_pts, img_pts, cam_matrix, distCoeffs, rodrigues_rvec, obj_T_cam_tvec, false, CV_ITERATIVE);
+    cv::Rodrigues(rodrigues_rvec, obj_T_cam_rvec); //rotational matrix conversion
+    
+    Eigen::MatrixXd obj_T_cam(4, 4);
+    obj_T_cam << obj_T_cam_rvec.at<double>(0, 0), obj_T_cam_rvec.at<double>(0, 1), obj_T_cam_rvec.at<double>(0, 2),
+        obj_T_cam_tvec.at<double>(0, 0), obj_T_cam_rvec.at<double>(1, 0), obj_T_cam_rvec.at<double>(1, 1),
+        obj_T_cam_rvec.at<double>(1, 2), obj_T_cam_tvec.at<double>(0, 1), obj_T_cam_rvec.at<double>(2, 0),
+        obj_T_cam_rvec.at<double>(2, 1), obj_T_cam_rvec.at<double>(2, 2), obj_T_cam_tvec.at<double>(0, 2), 0, 0, 0, 1;
+    
+    return obj_T_cam.inverse().eval();
+  }
+
+  //Calculates w_T_tag for all unknown tags in a file
+  void tagCalc(int filenum, int loc, bool world_pres)
+  {
+    YAML::Node tags_pix = YAML::LoadFile("detections_" + std::to_string(filenum) + ".yaml");
+    Eigen::MatrixXd w_T_cam(4, 4);
+    if(world_pres==0)
```

Comment:
Nit: My preference but i would suggest for booleans just use either True or False.